File: JAVA/Practica8/D.java
package JAVA.Practica8;
public class D extends A {
    private B b;
    public D(int x, int y, int z) {
        super(x);
        this.b = new B(y);
        this.z = z;
        this.b.z = z;
    }
    public int getY() {
        return b.y;
    }
    public int getZFromB() {
        return b.z;
    }
    public void incrementaYZ() {
        b.incrementaYZ();
    }
    public void incrementaXYZ() {
        x++;
        b.y++;
        z++;
        b.z++;
    }
    @Override
    public String toString() {
        return "x: " + x + ", y: " + b.y + ", z (from A): " + z + ", z (from B): " + b.z;
    }
}

File: PYTHON/Practica8/D.py
from A import A
from B import B

class D(A, B):
    def __init__(self, x, y, z):
        A.__init__(self, x)
        B.__init__(self, y)
        self.z = z
    def incrementaXYZ(self):
        self.x += 1
        self.y += 1
        self.z += 1
    def __str__(self):
        return f"x: {self.x}, y: {self.y}, z: {self.z}"